overlord/snapshotstate/snapshotstate.go

```diff
@@ -69,6 +73,70 @@ func allActiveSnapNames(st *state.State) ([]string, error) {
 	return names, nil
 }
 
+func automaticSnapshotExpiration(st *state.State) time.Duration {
+	// TODO: get from config
+	return DefaultAutomaticSnapshotExpiration
+}
+
+// saveExpiration saves expiration date of the given snapshot set in the state.
+// The state needs to be locked by the caller.
+func saveExpiration(st *state.State, setID uint64, expiryTime time.Time) error {
+	// note, expirations actually map to time.Time, use string here as we don't need parsed times here.
+	var expirations map[uint64]string
+	err := st.Get("snapshots-expiry", &expirations)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	if expirations == nil {
+		expirations = make(map[uint64]string)
+	}
+	expirations[setID] = expiryTime.Format(time.RFC3339)
+	st.Set("snapshots-expiry", expirations)
+	return nil
+}
+
+// removeExpirations removes expirations of the given set IDs from the state.
+func removeExpirations(st *state.State, setIDs ...uint64) error {
+	// note, expirations actually map to time.Time, use string here as we don't need parsed times here.
+	var expirations map[uint64]string
+	err := st.Get("snapshots-expiry", &expirations)
+	if err != nil {
+		if err == state.ErrNoState {
+			return nil
+		}
+		return err
+	}
+
+	for _, setID := range setIDs {
+		delete(expirations, setID)
+	}
+
+	st.Set("snapshots-expiry", expirations)
+	return nil
+}
+
+// expiredSnapshotSets returns a list of expired snapshot sets from the state, based on the given cutoffTime.
+// The state needs to be locked by the caller.
+func expiredSnapshotSets(st *state.State, cutoffTime time.Time) ([]uint64, error) {
```

Comment:
```suggestion
func expiredSnapshotSets(st *state.State, cutOffTime time.Time) ([]uint64, error) {
```

Reply:
I disagree with this change FWIW. "cutoff" is a perfectly cromulent word, and `cutoffTime` reads correctly like "the time of cutoff". `cutOffTime` implies it's a "cut off time", with the emphasis now on the "off", so this is now "the time of cutting off". What're you cutting off now?

Reply:
@chipaca thanks for explanation! I'll restore it in the 3/4 PR and land this one while it's green.